src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -100,6 +100,12 @@ public SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredTyp
                 allActions = new HashSet<IEdmAction>(model.GetAvailableActions(entityType));
                 allFunctions = new HashSet<IEdmFunction>(model.GetAvailableFunctions(entityType));
             }
+            else if (structuredType != null)
```

Comment:
so, why do we still keep the if(...) clause.

Reply:
We populate the actions and function only if it is an entity type. The second if indirectly implies it is only for the complex type. 

---
In reply to: [250464529](https://github.com/OData/WebApi/pull/1738#discussion_r250464529) [](ancestors = 250464529)

Reply:
So, it seems if we only change the return for "GetAvailableActions() and GetAvailableFunctions()", we can use "structuredType" directly, don't need to compare whether it's entity or complex?

---
In reply to: [250717218](https://github.com/OData/WebApi/pull/1738#discussion_r250717218) [](ancestors = 250717218,250464529)